### DISCUSSION TITLE
Cloud Based Wallets, Agents and Cloud attached Storage

### DISCUSSION THREAD

**Discussion by davidejalexander** [openwallet-foundation/architecture-sig#71] [Ideas]
One of the main attraction of OWF was its commitment to Cloud and Device Wallets and Agents and we can see some initiatives and projects working on this.  I am wondering what the Architecture SIG can be doing to explore this in more detail given the increasing demand for cloud based wallets and cloud attached store such as Personal Data Stores (PDS).

### **Key considerations for us**
**Individuals can have an independent core Cloud Wallet, Agents and Storage** that can interoperate with device based Wallets in different ecosystems as well as verified attribute / credential providers and relying parties who want to consume content from an individuals Wallet and Cloud storage.  The Cloud Storage provides a longtitudinal record of persons life as defined in data, transactions made up of a wide range of data from authortative sources.

**Individuals are able to set policy about access by relying parties** in an equivalent of Card Holder Not present settings where there is an ongoing relationship in which the exchange of personal data in a variety of forms is needed under data sharing agreements on a persistent basis not just during a transaction and interaction event.

### Questions arising

1. How will this impact the OWF Architecture or Wallets 
2. How do we identify the key interoperability and points of synchronisation between core cloud wallet and device wallets

**Comment by tkuhrt** [comment]
Capture of discussion from December 9, 2024 meeting:
- Custodial wallets would be considered in scope.
- Delegation of authority to the cloud agents.
- Design should include where the user does not need to be onboarded.
- How do you authenticate, identify, and verify the person is the actual owner of the identity.
- Loss of data and how do we ensure that this does not happen.
- How compatible is this with the EU ARF given the device-bound identity?
- Synchronization between wallets, which is different than backup/restore.
- How do you do key binding for cloud storage? Non-key-bound credentials can be stored on cloud and fetched to wallet when needed.
- Proximity and wallet-to-wallet will require credential to be local to the device.
- Start with needs, challenges, and gaps which will allow us to identify what exists today vs. what needs to be implemented.
- Link Web Storage W3C - https://github.com/w3c/lws-ucs/issues and https://www.w3.org/groups/wg/lws/
- Ability to move to a self-custodial solution from a custodial solution (and vice-versa).
- Investigate storage on user device alongside the cloud (hybrid solution).
- Device-bound vs. non-device bound considerations. Device-bound keys are not necessarily tied to a device but could be a cloud HSM. Might want to refer to this as key-bound credentials instead.

**Comment by mydexadmin** [comment]
Cloud Wallets are an individual challenge not just enterprise. They need to back up their wallets and synchronise across wallets and storage.

- Back up Recovery Repository.
- Synchronisation between wallets using a repository to provision and synch up - two way updates between wallets.
- Some wallets might be installed in a car
- **CHOICE** needs to embedded in ecosystems about storage choices and interoperability across different ecosystems, wallets and apps, services people use.
- App and wallet fatigue is rising, lose a sense of coherence and understanding which wallet or app has what information. Lots of FERC for joining the dots across these wallets / apps
- Consumable accepted design patterns for connectivity and interoperability that is consistent experience for all wallets users
- Can link to authenticatprs
- Can reduce the dependency on native device wallets storage, SEM and easily provision
- Makes it easier to deploy white label OWF open source created wallets
- Adapters for different forms of cloud storage and use cases with specific performance profiles and provides a a picture of what it looks like.
- Independent storage of the individual
- Agency of the individual to make choices to connect services e.g. wallets and apps to it for two way exchange
- Ability to delegate access and accept delegations from others
- Interoperability across ecosystems, across wallets, agent and apps and services
- Easier selective disclosure and minting of verified credentials based on a composite of other verified credentials from different ecosystems
- Support internet enabled dumb phones are coming as will other agent type services e.g. Alexa devices, vehicles, toll payments
- Making use of wallets effortless and secure as they require very little set up to get going.
- Device based wallets are at greater risk of compromise due to theft of devices just opened to pay for something

**### Next Steps**

1. Add clarification and examples of what it looks like
2. Explain how to map UWRA to enable mapping of storage, recovery back up, synchronisation and delegated access.
3. Present API within UWRA Architecture

**Comment by mydexadmin** [comment]
### Cloud Wallets and Cloud Storage discussions with reference to UWRA
https://github.com/openwallet-foundation/architecture-sig/discussions/71

The group discussed cloud wallets and storage and the need for wallet synchronization, with Romek emphasizing the importance of recovery backups and wallet synchronization across devices. They explored the concept of a "shell wallet" that can be populated with different functionalities, which Romek demonstrated through his work on identity and universal wallet reference architecture. David noted the need for examples of wallet content secure storage and user record secure storage to help shape how people design their cloud-based wallets.

**Personal Data Stores (Cloud Storage) and Interoperability:**
The group discussed the concept of personal data stores (PDS) and wallet interoperability, with David explaining that PDS enables individuals to maintain control over their data through user agency and guardianship features. Romek emphasized the importance of keeping data stored in the cloud rather than on devices, particularly for "dumb phones" which are becoming increasingly popular, and explained that only necessary information should be downloaded to the device for specific tasks. The discussion highlighted how this approach allows for better security and efficiency, with Carol noting that this vision has been evolving over the past decade through practical implementation and learning from user needs.

**Digital Wallet Security Challenges:**
The group discussed the challenges of digital wallets and personal data security, particularly in public spaces. David shared an anecdote from a government hackathon where a citizen struggled to understand the concept of multiple digital wallets, highlighting the need to simplify the user experience. Romek and David observed that current digital wallets, while convenient, may not be secure, as they often leave devices unlocked during transactions, making them vulnerable to theft. The conversation concluded with a discussion on the increasing reliance on digital wallets, especially among younger people, and the potential risks associated with storing sensitive information on devices that are easily stolen or compromised.

**Universal Wallet Reference Architecture Development:**
The team discussed the development of a universal wallet reference architecture and the Open Source project mapping and analysis tool being developed by David and Mydex . OSPMAE will be open sourced and will be used to provide the platform for OWF projects to be mapped and compared to aid in adoption.  It was agreed that we should consider the UWRA content relating to Cloud Wallets, Storage, interoperability, synchronisation, back up and recovery

Romek proposed a "pick and mix" approach where users would load specific wallet functions rather than carrying everything on their phone, making it more secure.

David emphasized the need to target the OWF projects that are already being adopted or close to adoption and to create a visual mapping tool that allows filtering and sorting of projects. 

**Cross referencing certification and testing requirements**
The team also discussed the need to include certifications, protocols, and standards in the wallet architecture, with Carol mentioning European Union requirements for accredited labs. Romek is working on implementing the architecture and creating examples for different components, while David is documenting the discussion points in GitHub.
